File: src/main/java/ua/softgroup/matrix/server/persistent/entity/TimeAudit.java
package ua.softgroup.matrix.server.persistent.entity;

import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.OneToOne;
import java.time.LocalDateTime;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
@Entity
public class TimeAudit extends AbstractEntity<Long> {
    private static final long serialVersionUID = 7093407748878141348L;

    @Column
    @CreationTimestamp
    private LocalDateTime creationDate;

    @Column
    private Integer addedMinutes;

    @Column
    private String reason;

    @OneToOne
    private User adder;

    @OneToOne
    private WorkDay workDay;

    public TimeAudit() {
    }

    public TimeAudit(Integer addedMinutes, String reason, User adder, WorkDay workDay) {
        this.addedMinutes = addedMinutes;
        this.reason = reason;
        this.adder = adder;
        this.workDay = workDay;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDateTime creationDate) {
        this.creationDate = creationDate;
    }

    public Integer getAddedMinutes() {
        return addedMinutes;
    }

    public void setAddedMinutes(Integer addedMinutes) {
        this.addedMinutes = addedMinutes;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public User getAdder() {
        return adder;
    }

    public void setAdder(User adder) {
        this.adder = adder;
    }

    public WorkDay getWorkDay() {
        return workDay;
    }

    public void setWorkDay(WorkDay workDays) {
        this.workDay = workDays;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/persistent/entity/TimeAudit.scala
package ua.softgroup.matrix.server.persistent.entity

import java.time.LocalDateTime
import javax.persistence.{Column, Entity, OneToOne}

import org.hibernate.annotations.CreationTimestamp
import org.springframework.data.jpa.domain.AbstractPersistable

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
@Entity
@SerialVersionUID(7093407748878141348L)
class TimeAudit extends AbstractPersistable[java.lang.Long] {

  @Column
  @CreationTimestamp
  @BeanProperty
  var creationDate: LocalDateTime = _

  @Column
  @BeanProperty
  var addedMinutes: Int = _

  @Column
  @BeanProperty
  var reason: String = _

  @OneToOne
  @BeanProperty
  var adder: User = _

  @OneToOne
  @BeanProperty
  var workDay: WorkDay = _

  def this(addedMinutes: Int, reason: String, adder: User, workDay: WorkDay) {
    this()
    this.addedMinutes = addedMinutes
    this.reason = reason
    this.adder = adder
    this.workDay = workDay
  }


  override def toString = s"TimeAudit(${super.getId}, creationDate=$creationDate, addedMinutes=$addedMinutes, reason=$reason)"
}
